src/environments/array.js

```diff
@@ -30,8 +30,18 @@ export type ArrayEnvNodeData = {
     // initialization.
     body?: ParseNode<*>[][], // List of rows in the (2D) array.
     rowGaps?: number[],
+    numHLinesBeforeRow?: number[],
 };
 
+function nextTokenIsHLine(parser: Parser): boolean {
+    parser.consumeSpaces();
+    if (parser.nextToken.text === "\\hline") {
+        parser.consume();
+        return true;
+    }
+    return false;
+}
```

Comment:
How about:
```js
function getNumHLines(parser: Parser): number {
    let n = 0;
    parser.consumeSpaces();
    while (parser.nextToken.text === "\\hline") {
        parser.consume();
        n++;
        parser.consumeSpaces();
    }
    return n;
}
```
(or `for` loop if you prefer) and

Reply:
I like it.